### DISCUSSION TITLE
Why does `hyprpm` not use installed headers

### DISCUSSION THREAD

**Discussion by cpiber** [hyprwm/Hyprland#6474] [General]
Currently, `hyprpm` always clones and builds a separate copy of Hyprland, even though the headers *must* be available somewhere if `hyprpm` has been built. I've patched mine to use the global headers and haven't had any trouble with it for a few weeks, why is this not standard?

**Comment by vaxerski** [comment]
we used to, but they were a lot of hassle and often would break due to user error.

**Comment by fufexan** [comment]
https://github.com/hyprwm/Hyprland/issues/6232#issuecomment-2161581384

**Reply by cpiber** [reply]
Honestly, if one just packages the installation of Hyprland without unnecessarily excluding headers etc this linked situation cannot happen. It also doesn't happen if one compiles for themselves, so this seems lacking as an argument

**Reply by fufexan** [reply]
My thoughts exactly. We should make a PR to hyprpm that checks pkg-config first.

**Reply by cpiber** [reply]
I could do such a thing, but not sure if @vaxerski would even accept it from their stance in the issue you linked

Though, is it even necessary to call pkg-config? cmake knows where the headers will be installed, so this can be added at compile-time, that also reduces the chance that the incorrect headers are found

**Reply by vaxerski** [reply]
liberum veto

**Reply by cpiber** [reply]
That's a no I guess :sweat_smile:
